src/commands/Minion/farmingcontract.ts

```diff
@@ -0,0 +1,179 @@
+import { getPlantToGrow } from '../../lib/skilling/functions/calcFarmingContracts';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import itemID from '../../lib/util/itemID';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			altProtection: true,
+			oneAtTime: true,
+			cooldown: 1,
+			usage: '<contractLevel:...string>',
+			usageDelim: ' ',
+			aliases: ['fc']
+		});
+	}
+
+	@minionNotBusy
+	@requiresMinion
+	async run(
+		msg: KlasaMessage,
+		[contractLevel]: ['completed' | 'easy' | 'medium' | 'hard' | 'easier' | 'current']
+	) {
+		await msg.author.settings.sync(true);
+		const farmingLevel = msg.author.skillLevel(SkillsEnum.Farming);
+		const currentContract = msg.author.settings.get(
+			UserSettings.FarmingContracts.FarmingContract
+		);
+
+		const userBank = msg.author.settings.get(UserSettings.Bank);
+
+		if (
+			contractLevel !== 'easy' &&
+			contractLevel !== 'medium' &&
+			contractLevel !== 'hard' &&
+			contractLevel !== 'easier' &&
+			contractLevel !== 'completed' &&
+			contractLevel !== 'current'
+		) {
+			return msg.send(
+				await guildmasterJaneImage(
+					`Are you a melon? Say 'easy', 'medium', or 'hard' for a contract. Say 'current' to see your current contract. Say 'easier' for an easier contract level. Say 'completed' to see how many contracts you've completed.`
+				)
+			);
+		}
+
+		if (contractLevel === 'completed') {
+			if (currentContract.contractsCompleted > 0) {
+				return msg.send(
+					await guildmasterJaneImage(
+						`I have checked my diary and you have completed a total of ${currentContract.contractsCompleted} farming contracts!`
+					)
+				);
+			}
+			return msg.send(
+				await guildmasterJaneImage(
+					`I tried to check my diary but you haven't completed any farming contracts yet...`
+				)
+			);
+		}
+
+		if (bankHasItem(userBank, itemID('Seed pack'), 1)) {
+			return msg.send(
+				await guildmasterJaneImage(
+					`You need to open your seed pack before receiving a new contract!`
+				)
+			);
+		}
+
+		if (currentContract.contractStatus && contractLevel === 'easier') {
+			return msg.send(
+				await guildmasterJaneImage(
+					`You currently don't have a contract, so you can't ask for something easier!`
+				)
+			);
+		}
+
+		if (contractLevel === 'easy' && farmingLevel < 45) {
+			return msg.send(
+				await guildmasterJaneImage(`You need 45 farming to receive an easy contract!`)
+			);
+		}
+		if (contractLevel === 'medium' && farmingLevel < 65) {
+			return msg.send(
+				await guildmasterJaneImage(`You need 65 farming to receive a medium contract!`)
+			);
+		}
+		if (contractLevel === 'hard' && farmingLevel < 85) {
+			return msg.send(
+				await guildmasterJaneImage(`You need 85 farming to receive a hard contract!`)
+			);
+		}
```

Comment:
this could be done a lot cleaner by mapping over  a data structure, rather than repeating the same thing 3 times.